Added Item Placement Marker

gccalvin opened the pull request:
Added the following feature:
- When dragging items, a red vertical line will display where the item (image) will be placed.

Please let me know of any suggested changes, modifications, or necessary cleanup.

gccalvin committed:
Added Item Placement Marker

When dragging items, a red vertical line will display where the item will be placed.

gccalvin committed:
Revert "Added Item Placement Marker"

This reverts commit 6827aca873cabdfee1beb6888653a3ce7816b537.

gccalvin committed:
Added Item Placement Marker

When dragging items, a red vertical line will display where the item will be placed.

silverweed commented:
I really like this feature! However I find it a bit confusing that the red line appears on the very left of the row (more left than the tier name) instead of at the end of the current images list when you want to append an image to the row.

Not sure if this is a problem with my browser, but here is what I mean:
![image](https://github.com/user-attachments/assets/48883506-9631-455e-ae4f-484613e03145)

It works perfectly fine when you drag an image anywhere except at the very end of the image list though.

silverweed commented on tiers.js:
```diff
@@ -268,6 +271,19 @@ function get_item_index(elem) {
 	return null;
 }
 
+// Sets the item placement marker render location
```
This comment is superfluous: please remove it

gccalvin committed:
Item Placement Marker Bug Fixes

Fixed various bugs with the placement marker:
1. The marker correctly shows at the end of the row when hovering a row.
2. The marker will no longer show on other UI elements, such as the '+' and row titles.
3. You can now move items within the bottom images container. The marker is correctly displayed.

silverweed commented on tiers.js:
```diff
@@ -265,9 +267,41 @@ function get_item_index(elem) {
+	h_offset -= 8;
+	if (is_hovering_row){
+		// Moves the vertical line to the right
+		placement_marker_div.style.marginLeft = (h_offset += 100) + "px";
```
Please put the `+= 100` in its own line

silverweed commented on tiers.js:
```diff
@@ -265,9 +267,41 @@ function get_item_index(elem) {
+	if (is_hovering_row){
+		// Moves the vertical line to the right
+		placement_marker_div.style.marginLeft = (h_offset += 100) + "px";
+	}
```
place the `else` on the same line as the closing brace (like this: `} else {`)

silverweed commented on tiers.js:
```diff
@@ -265,9 +267,41 @@ function get_item_index(elem) {
+		placement_marker_div.style.marginLeft = h_offset + "px";
+	}
+
+	var v_offset = elem.offsetTop.toString();
+	placement_marker_div.style.top = v_offset + "px";
```
```suggestion
	placement_marker_div.style.top = `${elem.offsetTop}px`;
```

silverweed reviewed (changes requested):
Very nice, now it works fine in all cases! Just a couple of minor changes and we can merge :+1:

gccalvin committed:
Formatting and Additional Bug Fixes

Applied suggested changes and fixed a few bugs.
- Fixed: The placement marker is incorrectly displayed when hovering over an empty bottom-container.
- Fixed: The placement marker is incorrectly displayed when hovering over the row labels or headers.
- Fixed: The placement marker is not removed when the drag ends on an un-droppable location (element).

gccalvin commented:
I have applied @silverweed's suggested changes and fixed some more bugs:
- Fixed: The placement marker is incorrectly displayed when hovering over an empty bottom-container.
![Screenshot 2025-04-05 111323](https://github.com/user-attachments/assets/1b96792f-f90f-4229-ab92-307b07b85443)

- Fixed: The placement marker is incorrectly displayed when hovering over the row labels or headers.
![Screenshot 2025-04-05 111223](https://github.com/user-attachments/assets/597b1f3e-887a-4e34-8d63-02e2db6dba44)

- Fixed: The placement marker is not removed when the drag ends on an un-droppable location (element).

Please review the new code I've added and let me know if any changes are needed. Thank you.

silverweed reviewed (approved):
Thank you!
